contracts/core/Allo.sol

```diff
@@ -10,11 +10,27 @@ import "@solady/auth/Ownable.sol";
 
 import {Metadata} from "./libraries/Metadata.sol";
 import {Clone} from "./libraries/Clone.sol";
-import "./libraries/Transfer.sol";
-import "../interfaces/IAllocationStrategy.sol";
-import "../interfaces/IDistributionStrategy.sol";
-import "./Registry.sol";
-
+import {Transfer} from "./libraries/Transfer.sol";
+import {BaseStrategy} from "../strategies/BaseStrategy.sol";
+import {Registry} from "./Registry.sol";
+
+/**
+ *          ___           ___       ___       ___
+ *         /\  \         /\__\     /\__\     /\  \
+ *        /::\  \       /:/  /    /:/  /    /::\  \
+ *       /:/\:\  \     /:/  /    /:/  /    /:/\:\  \
+ *      /::\~\:\  \   /:/  /    /:/  /    /:/  \:\  \
+ *     /:/\:\ \:\__\ /:/__/    /:/__/    /:/__/ \:\__\
+ *     \/__\:\/:/  / \:\  \    \:\  \    \:\  \ /:/  /
+ *          \::/  /   \:\  \    \:\  \    \:\  /:/  /
+ *          /:/  /     \:\  \    \:\  \    \:\/:/  /
+ *         /:/  /       \:\__\    \:\__\    \::/  /
+ *         \/__/         \/__/     \/__/     \/__/
+ */
```

Comment:
**_Nice!_**
I already planned to introduce something similar :D

Reply:
LMAO